### PR TITLE
SQLite dialect

### PR DESCRIPTION
**Pull request by luca-vercelli**
As discussed [here](https://github.com/hibernate/hibernate-orm/pull/1323), I have updated gradle configuration for including SQLite dialect.
Out of 369 tests, 28 fails and 42 are ignored, with a 91% of success. I guess tests fail due to JDBC limitations. Here is the list of failing tests:

    BatchTest. testScroll
    BatchTest. testStatelessSession
    AutoFlushTest. testFlushAutoSQL
    AutoFlushTest. testFlushAutoSQLNativeSession
    AutoFlushTest. testFlushAutoSQLSynchronization
    HQLTest. test_hql_all_subquery_comparison_qualifier_example
    HQLTest. test_hql_api_scroll_open_example
    HQLTest. test_hql_collection_expressions_example_5
    HQLTest. test_hql_collection_expressions_example_7
    HQLTest. test_hql_collection_expressions_example_8
    HQLTest. test_hql_extract_function_example
    HQLTest. test_hql_in_predicate_example_6
    HQLTest. test_hql_locate_function_example
    HQLTest. test_hql_numeric_arithmetic_example_2
    HQLTest. test_hql_numeric_arithmetic_example_3
    HQLTest. test_hql_substring_function_example
    HQLTest. test_hql_year_function_example
    BlobByteArrayTest. test
    BlobTest. test
    ClobCharArrayTest. test
    ClobStringTest. test
    ClobTest. test
    NClobCharArrayTest. test
    NClobStringTest. test
    NClobTest. test
    NationalizedTest. test
    DatabaseValueGenerationTest. test
    PooledOptimizerTest. test

### CODE REVIEW

**Comment by vladmihalcea** [comment]
Thanks @luca-vercelli. I'll review on Monday.

**Comment by vladmihalcea** [comment]
Thanks for the Pull Request. Did you sign the [CLA](https://cla.jboss.org/)?

**Comment by vladmihalcea** [comment]
You have a merge commit which prevents me from rebasing your work on top of master. Try squashing all your changes and submit it as one commit so that I can integrate it. We never issue any merge between branches. We always rebase.

File: hibernate-core/src/main/java/org/hibernate/dialect/SQLiteDialect.java [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+	// DDL support ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+	@Override
+	public boolean canCreateSchema() {
+		return false;
+	}
+
+	@Override
+	public boolean hasAlterTable() {
+		// As specified in NHibernate dialect
```
Comment (vladmihalcea) [inline hibernate-core/src/main/java/org/hibernate/dialect/SQLiteDialect.java]: Is this imported from NHibernate? Can we check it how it works there too?

File: hibernate-core/src/main/java/org/hibernate/dialect/identity/SQLiteDialectIdentityColumnSupport.java [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  /*
+	public boolean supportsInsertSelectIdentity() {
+    return true; // As specified in NHibernate dialect
+  }
+  */
+
+	@Override
+	public boolean hasDataTypeInIdentityColumn() {
+		// As specified in NHibernate dialect
+		// FIXME true
```
Comment (vladmihalcea) [inline hibernate-core/src/main/java/org/hibernate/dialect/identity/SQLiteDialectIdentityColumnSupport.java]: How should this be fixed in future. Is this also something related to NHibernate?

File: build.gradle [hunk 3]
```diff
@@ -169,6 +169,12 @@ subprojects { subProject ->
 		testRuntime( libraries.mssql )
 		testRuntime( libraries.informix )
 
+		if (db.equalsIgnoreCase("sqlite")) {
+			dependencies {
+				testRuntime( libraries.sqlite )
```
Comment (vladmihalcea) [inline build.gradle]: Why not providing it by default? The Oracle is a special case because it's not available on Maven Central.

**Comment by vladmihalcea** [comment]
@luca-vercelli I see you committed on master on your fork. This is not usually a good thing to do since it restricts you to fetching the latest changes from upstream.

What you can do to easily fix the merge commit problem is to follow these steps:

1. Create a new branch called `sqlite`
2. git checkout sqlite
3. git merge --squash master

Then you should have all changes in the `sqlite` branch which you can use to send the PR.

The master branch should be reset so that all extra commits you've done are removed. Then you can merge from upstream/master, assuming you added the upstream mirror, and then you can also rebase the `sqlite` branch.

**Comment by luca-vercelli** [comment]
@vladmihalcea, yes, I just begun to work on this. I will create a new branch ASAP.

**Comment by vladmihalcea** [comment]
@luca-vercelli Thanks

**Comment by luca-vercelli** [comment]
Dear @vladmihalcea your suggestion does not work. I did:
```
$git branch sqlite
$git checkout sqlite
$git merge --squash master
 (nothing to squash)Already up-to-date.
```

**Comment by luca-vercelli** [comment]
Maybe I could just drop completely my repository, then create a new one again and put all gwenn's modified files as they were mine... I don't like that so much...

**Comment by vladmihalcea** [comment]
You can also use `git cherry-pick` each commit that you did on master onto the new branch. Make sure that you skip the merge commits.

**Comment by luca-vercelli** [comment]
You should find a new branch sqlite. I have git cherry-picked all commits but merge commits into that branch.

**Comment by vladmihalcea** [comment]
Thanks. I'll check it out on Monday.

**Comment by vladmihalcea** [comment]
Replaced by this [PR](https://github.com/hibernate/hibernate-orm/pull/1874).

**Pull request closed without merging**